core/trino-main/src/main/java/io/trino/sql/planner/ConnectorExpressionTranslator.java

```diff
@@ -149,6 +220,62 @@ public SqlToConnectorExpressionTranslator(Map<NodeRef<Expression>, Type> types)
             return Optional.of(new Constant(null, typeOf(node)));
         }
 
+        @Override
+        protected Optional<ConnectorExpression> visitFunctionCall(FunctionCall node, Void context)
+        {
+            if (!isComplexExpressionPushdown(session)) {
+                return Optional.empty();
+            }
+
+            // Currently, we treat node as a builtin function.
+            // TODO Distinguish between builtin, plugin-provided and runtime-added functions.
+            if (node.getFilter().isPresent() || node.getOrderBy().isPresent() || node.getWindow().isPresent() || node.getNullTreatment().isPresent() || node.isDistinct()) {
+                return Optional.empty();
+            }
+
+            String functionName = ResolvedFunction.extractFunctionName(node.getName());
+
+            if (LiteralFunction.LITERAL_FUNCTION_NAME.equalsIgnoreCase(functionName)) {
+                Object value = ExpressionUtils.evaluateConstantExpression(plannerContext, session, node);
+                verify(!(value instanceof Expression), "Literal function did not evaluate to a constant: %s", node);
+                if (value instanceof JoniRegexp) {
+                    Slice pattern = ((JoniRegexp) value).pattern();
+                    return Optional.of(new Constant(pattern, VarcharType.createVarcharType(countCodePoints(pattern))));
+                }
+                if (value instanceof Re2JRegexp) {
+                    Slice pattern = Slices.utf8Slice(((Re2JRegexp) value).pattern());
+                    return Optional.of(new Constant(pattern, VarcharType.createVarcharType(countCodePoints(pattern))));
+                }
+                return Optional.of(new Constant(value, types.get(NodeRef.of(node))));
+            }
+
+            ImmutableList.Builder<ConnectorExpression> arguments = ImmutableList.builder();
+            for (Expression argumentExpression : node.getArguments()) {
+                Optional<ConnectorExpression> argument = process(argumentExpression);
+                if (argument.isEmpty()) {
+                    return Optional.empty();
+                }
+                arguments.add(argument.get());
+            }
+
+            return Optional.of(new Call(typeOf(node), Optional.empty(), functionName, arguments.build()));
+        }
+
+        @Override
+        protected Optional<ConnectorExpression> visitLikePredicate(LikePredicate node, Void context)
+        {
+            // TODO Support ESCAPE character
+            if (node.getEscape().isEmpty()) {
+                Optional<ConnectorExpression> value = process(node.getValue());
+                Optional<ConnectorExpression> pattern = process(node.getPattern());
+                if (value.isPresent() && pattern.isPresent()) {
+                    Type columnType = typeOf(node.getValue());
+                    return Optional.of(new Call(columnType, Optional.empty(), LIKE_PATTERN_FUNCTION_NAME, List.of(value.get(), pattern.get())));
```

Comment:
`columnType` is wrong. LIKE returns a BOOLEAN.

Reply:
Fixed